### ISSUE TITLE
Request ability to enable Istio support

### ISSUE THREAD

**Issue by hmklabs-dclow** [datawire/ambassador-chart#85]
Please describe your use case / problem.
As a user of Ambassador and Istio I would like to be able to deploy Ambassador with Istio support enabled using the Ambassador Helm chart.

Describe the solution you'd like
It would be helpful if there was a configuration option in the Helm chart that would add the configuration for the Istio sidecar into the Ambassador installation.

Describe alternatives you've considered
There is a recent document on the steps to enable Istio support. However, this would need to be configured outside the Helm chart.

https://www.getambassador.io/docs/latest/howtos/istio/

**Comment by nicolasbelanger** [comment]
I'm having the same issue.  [Helm post-rendering](https://helm.sh/docs/topics/advanced/#post-rendering) looks like a good way out.  Here's an [example](https://github.com/thomastaylor312/advanced-helm-demos/tree/master/post-render).  Basically extracting a patch out of the [doc](https://www.getambassador.io/docs/latest/howtos/istio/) and enjoy (works in my head at least :p )

I haven't tried it yet, but will do soon :)

**Comment by kmattiasek1** [comment]
+1 on this feature

**Comment by nicolasbelanger** [comment]
If that can help, I've been able to push what's needed without touching anything on the chart itself, everything as a config.  Basically you need to have the sidecar injected by istio.

Versions tested:
**Istio**: 1.6.8
**Ambassador**: 1.6.2

Ref.: https://github.com/datawire/ambassador/issues/2587#issuecomment-676587914

TL;DR
```bash
kubectl label namespace <ambassador_namespace_name> istio.io/rev=<rev_name>
```

Edit `values.yaml` to include:
```yaml
env:
  AMBASSADOR_ENVOY_BASE_ID: "1"

podAnnotations:
  proxy.istio.io/config: |
    proxyMetadata:
      OUTPUT_CERTS: /etc/istio-certs
  sidecar.istio.io/userVolume: '[{"name": "istio-certs", "emptyDir": {"medium": "Memory"}}]'
  sidecar.istio.io/userVolumeMount: '[{"name": "istio-certs", "mountPath": "/etc/istio-certs"}]'
  traffic.sidecar.istio.io/includeInboundPorts: ''
  traffic.sidecar.istio.io/includeOutboundIPRanges: ''
  
volumes:
  - name: istio-certs
    emptyDir:
      medium: Memory

volumeMounts:
  - mountPath: /etc/istio-certs/
    name: istio-certs
```

There is still the fact that the mounted certs are not (yet) refreshed on expiration/istio rotation.
